lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java

```diff
@@ -188,4 +191,46 @@ public FileIterator listFiles(Location location)
             }
         });
     }
+
+    @Override
+    public Optional<Boolean> directoryExists(Location location)
+            throws IOException
+    {
+        stats.getDirectoryExistsCalls().newCall();
+        Path directory = hadoopPath(location);
+        FileSystem fileSystem = environment.getFileSystem(context, directory);
+
+        if (!hierarchical(fileSystem, location)) {
```

Comment:
This needs to be inside the `doAs()` block below, otherwise it can fail when using HDFS authentication